lib/ClangImporter/ImportDecl.cpp

```diff
@@ -5636,6 +5660,77 @@ Decl *SwiftDeclConverter::importGlobalAsInitializer(
   return result;
 }
 
+Decl *SwiftDeclConverter::importCXXMethodDecl(
+    const clang::CXXMethodDecl *decl, DeclName name, DeclContext *dc,
+    Optional<ImportedName> correctSwiftName) {
+  auto isStatic = decl->isStatic();
+  auto loc = Impl.importSourceLoc(decl->getLocation());
+  if (dc->getSelfProtocolDecl() && isStatic) {
+    Impl.SwiftContext.Diags.diagnose(loc, diag::swift_name_protocol_static,
+                                     /*isInit=*/false);
+    Impl.SwiftContext.Diags.diagnose(loc, diag::note_while_importing,
+                                     decl->getName());
+    return nullptr;
+  }
+
+  if (!decl->hasPrototype()) {
+    Impl.SwiftContext.Diags.diagnose(loc, diag::swift_name_no_prototype);
+    Impl.SwiftContext.Diags.diagnose(loc, diag::note_while_importing,
+                                     decl->getName());
+    return nullptr;
+  }
+
+  bool allowNSUIntegerAsInt =
+      Impl.shouldAllowNSUIntegerAsInt(isInSystemModule(dc), decl);
+
+  auto &C = Impl.SwiftContext;
+  auto *bodyParams = Impl.importFunctionParameterList(
+      dc, decl, {decl->param_begin(), decl->param_end()}, decl->isVariadic(),
+      allowNSUIntegerAsInt, name.getArgumentNames());
+
+  auto importedType =
+      Impl.importFunctionReturnType(dc, decl, allowNSUIntegerAsInt);
+  Type swiftResultTy = importedType.getType();
+
+  auto nameLoc = Impl.importSourceLoc(decl->getLocation());
+  auto result = createFuncOrAccessor(C, loc, None, name, nameLoc, bodyParams,
+                                     swiftResultTy,
+                                     /*throws*/ false, dc, decl);
+
+  result->setGenericEnvironment(dc->getGenericEnvironmentOfContext());
+
+  result->setAccess(AccessLevel::Public);
+  // Workaround until proper const support is handled: Force everything to be
+  // Mutating. This implicitly makes the parameter indirect. There may be
+  // contexts.
+  result->setSelfAccessKind(SelfAccessKind::Mutating);
```

Comment:
This should only be if `!isStatic`, since static member functions don't have a `this` parameter.

Reply:
Yes, of course.